bin/node-template/node/src/cli.rs

```diff
@@ -9,3 +11,21 @@ pub struct Cli {
 	#[structopt(flatten)]
 	pub run: RunCmd,
 }
+
+pub struct Adapter;
+
+impl RuntimeAdapter for Adapter {
+	type Runtime = Runtime;
+	type Extra = SignedExtra;
+
+	fn build_extra(index: Index) -> Self::Extra {
+		(
+			frame_system::CheckVersion::new(),
+			frame_system::CheckGenesis::new(),
+			frame_system::CheckEra::from(Era::Immortal),
```

Comment:
I think this should come from the cli, but not sure what that should look like from the cli side, @gnunicorn thoughts?

Reply:
we might want to consider `RuntimeAdapter` something to be implemented directly on `Runtime`. This is very much runtime dependent, and looks a lot like the `TransacationSubmitter` – maybe we can combine those, @tomusdrw !?!

Reply:
Let's make a first attempt in this PR and get that merged. Let's tackle https://github.com/paritytech/substrate/issues/5330 on a wider angle with the other parties involved after.

Reply:
I'm not in favor of merging this as it as it will block my work on the sc-cli part (if we do want to move RuntimeAdapter out of sc-cli)

Reply:
sorry, I might have phrased that badly. I meant: let's take a crack at what I proposed, just not wait for 5330 – and be aware we will clean that approach up soon.